### PR TITLE
fix: handle WouldBlock and read in loop

### PR DESCRIPTION
**Pull request by ForsakenHarmony**
Don't know if it's a windows only issue, but it wouldn't start without throwing an error, and now it works fine

### CODE REVIEW

**Comment by ForsakenHarmony** [comment]
nvm, still broken

**Comment by kelnos** [comment]
Hm, I definitely don't see this problem on Linux... unfortunately I just don't know much about how this works on Windows.  If you have time, could you try with 0.3.0?  That would be before I switched from a simple `.read()` loop, to using `poll()` with `mio`.  Not sure if that will be better, but it might tell us something.

Unfortunately I deleted my Windows partition a year or so ago, so I can't test on my own.

I ended up using `mio` because I didn't feel like going to async/await with `tokio` (and needed to start polling two read sources to handle stdin), but maybe I should have just bit that particular bullet.

**Comment by ForsakenHarmony** [comment]
0.3.0 does seem to work
0.4.0 with this change prints out some of the serial output and then just stops

**Comment by ForsakenHarmony** [comment]
Ok, looked at a mio example and they had a loop, added it here as well, that seems to fix it

File: espmonitor/src/lib.rs [hunk 1]
```diff
@@ -348,6 +350,7 @@ fn process_line(reader: &SerialReader, line: &str) -> String {
                 .ok()
                 .and_then(|output| String::from_utf8(output.stdout).ok())
             {
+                println!("output: {}", output);
```
Comment (kelnos) [inline espmonitor/src/lib.rs:353]: I assume this is some leftover debugging and should be removed?

**Review by kelnos — CHANGES_REQUESTED** [review]
Ah, makes sense.  No guarantee that `.read()` will read everything that's available, even if the buffer is large enough, and I guess `poll()` probably won't notify again for data it's already notified about.  Thanks so much for looking into this!  Just one small thing...

File: espmonitor/src/lib.rs (hunk 1 above)
Comment (ForsakenHarmony) [reply espmonitor/src/lib.rs:353]: yes

**Review by kelnos — APPROVED** [review]
Great, thanks!

**Comment by kelnos** [comment]
Released 0.4.1 with your fix.  Thanks again!

**Comment by ForsakenHarmony** [comment]
Thanks for the quick merge

**Pull request merged**